Calibration/IsolatedParticles/plugins/IsoTrackCalib.cc

```diff
@@ -611,52 +697,35 @@ double IsoTrackCalib::dEta(math::XYZTLorentzVector& vec1, math::XYZTLorentzVecto
 }
 
 double IsoTrackCalib::dPhi(math::XYZTLorentzVector& vec1, math::XYZTLorentzVector& vec2) {
-  return reco::deltaPhi(vec1.phi(),vec2.phi());
-}
 
-double IsoTrackCalib::dR(math::XYZTLorentzVector& vec1, math::XYZTLorentzVector& vec2) {
-  return reco::deltaR(vec1.eta(),vec1.phi(),vec2.eta(),vec2.phi());
+  double phi1 = vec1.phi();
+  if (phi1 < 0) phi1 += 2.0*M_PI;
+  double phi2 = vec2.phi();
+  if (phi2 < 0) phi2 += 2.0*M_PI;
+  double dphi = phi1-phi2;
+  if (dphi > M_PI)       dphi -= 2.*M_PI;
+  else if (dphi < -M_PI) dphi += 2.*M_PI;
+  return dphi;
 }
 
-double IsoTrackCalib::dPt(math::XYZTLorentzVector& vec1, math::XYZTLorentzVector& vec2) {
-  return (vec1.pt()-vec2.pt());
+double IsoTrackCalib::dR(math::XYZTLorentzVector& vec1, math::XYZTLorentzVector& vec2) {
+  double deta = dEta(vec1,vec2);
+  double dphi = dPhi(vec1,vec2);
+  return std::sqrt(deta*deta + dphi*dphi);
 }
 
-double IsoTrackCalib::dP(math::XYZTLorentzVector& vec1, math::XYZTLorentzVector& vec2) {
-  return (std::abs(vec1.r()-vec2.r()));
+double IsoTrackCalib::deltaR(double eta1,double eta2,double phi1,double phi2){
+  double deta = eta1-eta2;
+  if (phi1 < 0) phi1 += 2.0*M_PI;
+  if (phi2 < 0) phi2 += 2.0*M_PI;
+  double dphi = phi1-phi2;
+  if (dphi > M_PI)       dphi -= 2.*M_PI;
+  else if (dphi < -M_PI) dphi += 2.*M_PI;
+  return std::sqrt(deta*deta + dphi*dphi); 
```

Comment:
all these functions are already implemented in `DataFormats/Math`. Can you use the ones maintained by RECO?